service/controller/resource/workermigration/internal/azure/spec.go

```diff
@@ -0,0 +1,17 @@
+package azure
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+)
+
+type VMSS *compute.VirtualMachineScaleSet
```

Comment:
I read `VMSS` in a different file and I started wondering why you weren't using a pointer. Then I realized `VMSS` is an alias for a pointer. Wouldn't be more explicit to keep the alias to `type VMSS compute.VirtualMachineScaleSet` so other places would use `*VMSS` making it clear that it's actually a pointer? I don't know if there are any Go guidelines about this.

Reply:
Hmm... You have a point there. I was thinking about the most lightweight mechanism to make have an interface abstraction for Azure SDK types and this worked quite nicely without introduction of actual `interface` and all the boilerplate that comes with it.

I tried to change this but it seems that pointer alias gets "interface treatment" while struct alias actually creates a new type and it cannot be cast from one to another (because it's not an interface type).